fairlearn/reductions/_moments/utility_parity.py

```diff
@@ -88,26 +102,77 @@ def default_objective(self):
         """Return the default objective for moments of this kind."""
         return ErrorRate()
 
-    def load_data(self, X, y, event=None, utilities=None, **kwargs):
-        """Load the specified data into this object.
+    def compute_base_event(self, X: pd.DataFrame, y: pd.Series) -> pd.Series:
```

Comment:
What's the need for this? Why do we need this more complicated mechanism and cannot just handle everything on the level of `load_data`? if you prefer, you could change the signature to something like:
```python
def load_data(self, X, y, *, sensitive_features, event=None, utilities=None, control_features=None)
```

Reply:
I came up with this pattern to address the following comment:
https://github.com/fairlearn/fairlearn/pull/631#discussion_r540527969
That version of the code left `UtilityParity` unchanged, which meant that the code to handle `control_features` was appearing in each `load_data()` method.

Reply:
I see. I think that a better way to avoid copy-pasting would be to go essentially with your previous implementation, but create a helper function to handle combining `base_event` and `control_features` into `event`.

The basic issue that I see with this proposal is that `UtilityRate` should not be dealing with `control_features` at all. (In fact, it shouldn't be even dealing with `y`, but that's for another PR.)

Reply:
Done